tests/run_test.cmake

```diff
@@ -1,113 +1,91 @@
 #
-# This is a wrapper cmake script to run one uncrustify test because cmake add_test
-# can't launch multiple processes and so it doesn't allow a complex test scenario.
+# This is a wrapper CMake script to run one uncrustify test because CMake
+# add_test can't launch multiple processes on its own, and so doesn't directly
+# allow a complex test scenario.
 #
 
 cmake_minimum_required(VERSION 2.8)
 
+function(require)
+  foreach(var IN LISTS ARGN)
+    if(NOT ${var})
+      message(FATAL_ERROR "Variable ${var} not defined")
+    endif()
+  endforeach()
+endfunction()
 
-if(NOT TEST_PROGRAM)
-  message(FATAL_ERROR "Variable TEST_PROGRAM not defined")
-endif()
-if(NOT TEST_LANG)
-  message(FATAL_ERROR "Variable TEST_LANG not defined")
-endif()
-if(NOT TEST_CONFIG)
-  message(FATAL_ERROR "Variable TEST_CONFIG not defined")
-endif()
-if(NOT TEST_RERUN_CONFIG)
-  message(FATAL_ERROR "Variable TEST_RERUN_CONFIG not defined")
-endif()
-if(NOT TEST_INPUT)
-  message(FATAL_ERROR "Variable TEST_INPUT not defined")
-endif()
-if(NOT TEST_OUTPUT)
-  message(FATAL_ERROR "Variable TEST_OUTPUT not defined")
-endif()
-if(NOT TEST_RESULT)
-  message(FATAL_ERROR "Variable TEST_RESULT not defined")
-endif()
-if(NOT TEST_RESULT_2)
-  message(FATAL_ERROR "Variable TEST_RESULT_2 not defined")
-endif()
-if(NOT TEST_DIR)
-  message(FATAL_ERROR "Variable TEST_DIR not defined")
-endif()
-if(NOT TEST_RERUN_OUTPUT)
-  message(FATAL_ERROR "Variable TEST_RERUN_OUTPUT not defined")
-endif()
-
-
-# first pass
-execute_process(
-  COMMAND ${TEST_PROGRAM} -l ${TEST_LANG} -c ${TEST_CONFIG} -f ${TEST_INPUT} -o ${TEST_RESULT}
+function(run_test name config input result output)
```

Comment:
I'd like to swap result and output because in a test, usually the assert is `is_equal(expected, result)`.
And yeah, maybe output could be renamed to expected but that's another story.

Reply:
I *almost* did that ("output" → "expected", that is) :smile:, but left it "output" because that's what was already in use. I'll switch the order.